src/app/_components/profile/UserDetails/UserDetailsDisplay.tsx

```diff
@@ -5,49 +5,87 @@ import {
   const session = useSession();
 
-  const { data: userDetails } = api.user.findUserById.useQuery(
-    session.data?.user?.id ?? "",
-    { enabled: !!session.data },
-  );
+  const { data: userDetails, isLoading: isLoadingUserDetails } =
+    api.user.findUserById.useQuery(session.data?.user?.id ?? "", {
+      enabled: !!session.data,
+    });
 
   const facts = [
     `I live in ${userDetails?.city}`,
     `I am a ${userDetails?.gender}.`,
     `I am looking for a ${userDetails?.sexualPreference}!`,
   ];
+  const {
+    data: usersImages,
+    refetch: refetchImages,
+    isLoading: isLoadingImages,
+  } = api.image.findMany.useQuery(session.data?.user?.id ?? "", {
+    enabled: !!session.data,
+  });
 
   return (
     <Card className="max-w-screen-sm">
       <CardHeader>
-        <CardTitle>{`${userDetails?.name}, ${userDetails?.age}`}</CardTitle>
-        <CardDescription>{userDetails?.email}</CardDescription>
+        {isLoadingUserDetails || !userDetails ? (
+          <div className="space-y-2">
+            <Skeleton className="h-8 w-1/5 rounded-md" />
+            <Skeleton className="h-4 w-1/6 rounded-md" />
+          </div>
+        ) : (
+          <>
+            <CardTitle>{`${userDetails?.name}, ${userDetails?.age}`}</CardTitle>
+            <CardDescription>{userDetails?.email}</CardDescription>
+          </>
+        )}
       </CardHeader>
       <CardContent>
         <div className="flex justify-between">
-          <div>
-            {facts.map((fact, index) => (
-              <div
-                key={index}
-                className="mb-4 grid grid-cols-[25px_1fr] items-start pb-4 last:mb-0 last:pb-0"
-              >
-                <span className="flex h-2 w-2 translate-y-1 rounded-full bg-primary" />
-                <div className="space-y-1">
-                  <p className="text-sm font-medium leading-none">{fact}</p>
+          {isLoadingUserDetails || isLoadingImages || !session.data ? (
+            <div>
+              {facts.map((_, index) => (
+                <div
+                  key={index}
+                  className="mb-4 grid grid-cols-[25px_1fr] items-start pb-4 last:mb-0 last:pb-0"
+                >
+                  <Skeleton className="h-2 w-2 translate-y-1 rounded-full" />
+                  <div className="space-y-1">
+                    <Skeleton className="h-4 w-48" />
+                  </div>
+                </div>
+              ))}
+            </div>
+          ) : (
+            <div>
+              {facts.map((fact, index) => (
+                <div
+                  key={index}
+                  className="mb-4 grid grid-cols-[25px_1fr] items-start pb-4 last:mb-0 last:pb-0"
+                >
+                  <span className="flex h-2 w-2 translate-y-1 rounded-full bg-primary" />
+                  <div className="space-y-1">
+                    <p className="text-sm font-medium leading-none">{fact}</p>
+                  </div>
```

Comment:
maybe consider the if thing  here aswell if you think its right

Reply:
it is too much jsx so it'll be ugly